src/concat/concat-files.js

```diff
@@ -1,16 +1,30 @@
 var FileType = require('../file').type;
 var Combiner = require('./source-map-combiner');
 var Promise = require('bluebird');
+var requireify = require('./requireify');
 
 /**
  * @param {object} options
+ * @param {string} options.bundleName
  * @param {Array<object>} options.files
  * @param {string} options.fileType
  * @param {boolean} options.sourceMap
+ * @param {boolean} options.require
+ * @param {object} options.bundleStatsCollector
  * @returns {Promise}
  */
 function concat(options) {
 
+    if (options.require && options.fileType === FileType.JS) {

```

Comment:
When concatenating files in a bundle, if the `require` option is set, we use `requireify` rather than our old simple string concat method.